### PR TITLE
Fix codeblocks overflow

### LINKED ISSUE #87043: Rustdoc: long lines are escaping text code blocks
Repro: 

```rust
/// ```text
/// aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
/// ```
/// ```rust
/// bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
/// ```
pub struct Item;
```
Rustdoc output:
image

It seems that only code block with `text` language are affected.

cc @GuillaumeGomez

### Meta

```text
; rustdoc --version --verbose
rustdoc 1.55.0-nightly (240ff4c4a 2021-07-09)
binary: rustdoc
commit-hash: 240ff4c4a0d0936c9eeb783fa9ff5c0507a6ffb4
commit-date: 2021-07-09
host: x86_64-unknown-linux-gnu
release: 1.55.0-nightly
LLVM version: 12.0.1
```

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #87043.

Instead of completely relying on `pulldown-cmark` (and its potential changes), I decided to move the generation of codeblocks HTML directly in rustdoc so we can unify the DOM and the CSS classes.

r? @Nemo157

### CODE REVIEW

**Review by notriddle — APPROVED** [review]

File: src/librustdoc/html/markdown.rs [hunk 1]
```diff
@@ -241,6 +226,35 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for CodeBlocks<'_, 'a, I> {
...
+                let parse_result =
+                    LangString::parse_without_check(&lang, self.check_error_codes, false);
+                if !parse_result.rust {
+                    return Some(Event::Html(
+                        format!(
+                            "<div class=\"example-wrap\">\
+                                 <pre{}>{}</pre>\
+                             </div>",
+                            format!(" class=\"language-{}\"", lang),
+                            text,
```
Comment (notriddle) [inline src/librustdoc/html/markdown.rs:240]:
```suggestion
                            crate::html::escape::Escape(text),
```

Comment (GuillaumeGomez) [reply src/librustdoc/html/markdown.rs]: I completely forgot to escape it... Thanks!

Comment (notriddle) [reply src/librustdoc/html/markdown.rs]: That’s fine. I should’ve been looking for it.

**Pull request merged**